plugins/modules/ec2_instance.py

```diff
@@ -1567,9 +1567,9 @@ def build_network_spec(client, module: AnsibleAWSModule) -> List[Dict[str, Any]]
                 for inty in network_interfaces
             ]
         )
-    elif not network and not network_interfaces_ids:
-        # They did not specify any network interface configuration
-        # build network interface using subnet_id and security group(s) defined on the module
+    elif not network and not network_interfaces_ids and not module.params.get("launch_template"):
```

Comment:
I won't block the PR on this, but as far as I can tell, this is the only line that actually needs to be changed to fix the bug. I'd really prefer that any refactoring should happen in a separate PR.

Reply:
Handling the refactoring in a separate PR makes sense. I'll leave the decision to @abikouo , as he is the original author of this PR.

Reply:
@gravesm the PR does not include any code refactoring, all changes are aimed at fixing the reported issue